modules/internal/ChapelRange.chpl

```diff
@@ -2176,10 +2176,57 @@ operator :(r: range(?), type t: range(?)) {
   // An unbounded range iterator (for all strides)
   pragma "no doc"
   pragma "order independent yielding loops"
-  iter range.these() where boundedType != BoundedRangeType.bounded {
+  iter range.these() where boundedType == BoundedRangeType.boundedNone {
+    compilerError("iteration over a range with no bounds");
+  }
 
-    if boundedType == BoundedRangeType.boundedNone then
-      compilerError("iteration over a range with no bounds");
+  // An unbounded range iterator (for all strides)
+  pragma "no doc"
+  pragma "order independent yielding loops"
+  iter range.these() where boundedType == BoundedRangeType.boundedLow {
+
+    if boundsChecking {
+      if ! this.hasFirst() then
+        HaltWrappers.boundsCheckHalt("iteration over range that has no first index");
+
+      if this.isAmbiguous() then
+        HaltWrappers.boundsCheckHalt("these -- Attempt to iterate over a range with ambiguous alignment.");
+    }
+    // This iterator could be split into different cases depending on the
+    // stride like the bounded iterators. However, all that gets you is the
+    // ability to use low/alignedLow over first. The additional code isn't
+    // worth it just for that.
+    var i: intIdxType;
+    const start = chpl__idxToInt(this.first);
+    const end = if isBoolType(idxType)
+                  then 1: intIdxType
+                else if isEnumType(idxType)
+                  then (idxType.size-1):intIdxType
+                else max(intIdxType) - stride: intIdxType;
```

Comment:
Why is that we don't have to worry about stride for bool and enum types but have to for other types?

Reply:
> I guess we're representing enums as int(64) so the chance of under\overflowing is unlikely.

That's right.  Though even if we were representing them as int(32) or int(16), it'd still be unlikely.  Even int(8) is a pretty big enum, though not an inconceivably large one...

> we could get a bad interaction someday if we ever had our backend compiler use an optimization like this (-fshort-enum to optimize the type used to store enums to the type with the "shortest" satisfying range):

I don't think so because we're not exposing Chapel enums to the back-end compiler as enums.  Any enum-ness is handled in Chapel itself, and the back-end compiler just sees integers.